configure.py

```diff
@@ -1496,17 +1503,43 @@ def _generate_target_source_files_isa_properties(fd, target):
     @staticmethod
     def _write_header(fd):
         fd.write('cmake_minimum_required(VERSION 2.8.0)\n')
-        fd.write('project(botan LANGUAGES CXX)\n\n')
+        fd.write('project(botan)\n\n')
+        fd.write('if(POLICY CMP0042)\n\n')
+        fd.write('cmake_policy(SET CMP0042 NEW)\n\n')
         fd.write('cmake_policy(SET CMP0042 NEW)\n\n')
+        fd.write('endif()\n\n')
+        fd.write('function(to_cmake_paths paths results)\n')
+        fd.write('    set(_results)\n')
+        fd.write('	foreach(path ${paths})\n')
+        fd.write('		file(TO_CMAKE_PATH ${path} path)\n')
+        fd.write('		set (_results ${_results} ${path})\n')
+        fd.write('	endforeach()\n')
+        fd.write('	set(${results} ${_results} PARENT_SCOPE)\n')
+        fd.write('endfunction()\n\n')
 
     def _write_footer(self, fd, library_link, cli_link, tests_link):
         fd.write('\n')
 
         fd.write('option(ENABLED_OPTIONAL_WARINIGS "If enabled more strict warinig policy will be used" OFF)\n')
         fd.write('option(ENABLED_LTO "If enabled link time optimization will be used" OFF)\n\n')
-
-        fd.write('set(COMPILER_FEATURES %s %s)\n'
+        original_no_optimizations = self._options.no_optimizations
+        original_with_debug_info = self._options.with_debug_info
+        self._options.no_optimizations = False
```

Comment:
Changing self._options is not what you want to do here, because self._options is a map, which is a reference type in Python. So whenever you touch your reference of self._options, the entire configure script is affected. I know you revert the changes later but this is easily changed into bugs.

Let's take the safe route and create a local copy of the options that you can youse locally for whatever you want and throw away when you're done:

```python
my_local_options = copy.deepcopy(self._options)
```